tests/virt_autotest/vgpu.pm

```diff
@@ -0,0 +1,368 @@
+use ipmi_backend_utils qw(reconnect_when_ssh_console_broken);
+
+our $log_dir = "/tmp/vgpu";
+our $vm_xml_save_dir = "/tmp/download_vm_xml";
+
+sub run_test {
+    my $self = shift;
+
+    # Use sol & ssh console explixitly in case test module run on an installed host
+    # thus no login_console.pm has been required any more
+    select_console 'sol', await_console => 0;
+    use_ssh_serial_console;
+
+    # Test Prerequisite is fulfilled
+    return unless is_host_ready_for_vgpu_test() eq 'yes';
+
+    my $gpu_device = get_gpu_device();
+    die "No NVIDIA AMPERE GPU card on the machine!" if $gpu_device eq '';
+
+    # Clean up test logs
+    script_run "[ -d $log_dir ] && rm -rf $log_dir; mkdir -p $log_dir";
+
+    # Install NVIDIA vGPU manager
+    my $vgpu_manager = get_required_var("VGPU_MANAGER_URL");
+    install_vgpu_manager_and_reboot($vgpu_manager);
+
+    # Enable SR-IOV VFs
+    enable_sriov($gpu_device);
+
+    return if get_var('VGPU_SHORT_TEST');
+
+    # Create a CI for GPU with MIG mode supported, such A30/A100
+    # given MIG mode is enabled. will handle non MIG mode when zen2 is available
+    if (get_var('GPU_MIG_MODE')) {
+        enable_mig_mode();
+        # Create a GI
+        my ($gi_id, $gpu_id) = create_gpu_instance();
+        # Create a CI
+        my $ci_id = create_compute_instance($gi_id, $gpu_id);
+    }
+
+    # Create a vGPU device
+    my $vgpu = create_vgpu($gpu_device, get_var('GPU_MIG_MODE'));
+
+    save_original_guest_xmls();
+    my $vgpu_grid_driver = get_required_var("VGPU_GRID_DRIVER_URL");
+    foreach my $guest (keys %virt_autotest::common::guests) {
+        record_info("Guest $guest");
+
+        # Assign vGPU to guest
+        my $gpu_slot_in_guest = "0x0a";
+        assign_vgpu_to_guest($vgpu, $guest, $gpu_slot_in_guest);
+
+        # Install vGPU grid driver in guest
+        install_vgpu_grid_driver($guest, $vgpu_grid_driver);
+
+        # Detach vGPU from guest
+        detach_vgpu_from_guest($gpu_slot_in_guest, $guest);
+        check_guest_health($guest);
+    }
+
+    # Remove vGPU device
+    remove_vgpu($vgpu);
+
+    # Disable GPU SR-IOV
+    assert_script_run("/usr/lib/nvidia/sriov-manage -d $gpu_device");
+    record_info("SRIOV disabled", $gpu_device);
+
+    # Upload vgpu related logs
+    upload_virt_logs($log_dir, "logs");
+
+    # Redefine guest from their original configuration files
+    restore_original_guests();
+}
+
+sub is_host_ready_for_vgpu_test {
+    if (is_sle('<15-SP3') or !is_kvm_host) {
+        record_info("Host not supported!", "NVIDIA vGPU is only supported on KVM hosts with SLE15SP3 and greater!", result => 'softfail');
+        return;
+    }
+    #check VT-d is supported in Intel x86_64 machines
+    if (script_run("grep -m 1 Intel /proc/cpuinfo") == 0) {
+        assert_script_run "dmesg | grep -E \"DMAR:.*IOMMU enabled\"";
+    }
+    script_run("lsmod | grep -e vfio -e nvidia -e mdev");
+    return 'yes';
```

Comment:
It seems more common to return 1 for true case :)

Reply:
Yes, I just checked that majory boolean functions return 1 in our tests. I'll update.